receiver/otlpreceiver/otlp.go

```diff
@@ -77,11 +77,18 @@ func New(
 		return nil, fmt.Errorf("failed to bind to address %q: %v", addr, err)
 	}
 
+	protobufMarshaler := &xProtobufMarshaler{}
 	r := &Receiver{
 		ln:          ln,
 		corsOrigins: []string{}, // Disable CORS by default.
 		gatewayMux: gatewayruntime.NewServeMux(
-			gatewayruntime.WithMarshalerOption("application/x-protobuf", &xProtobufMarshaler{}),
+			gatewayruntime.WithMarshalerOption("application/x-protobuf", protobufMarshaler),
+			// Explicitly handle "application/json" encoding to return custom human friendly error
+			// See issue https://github.com/open-telemetry/opentelemetry-proto/issues/129
+			// describing why we don't support JSON at the moment.
+			gatewayruntime.WithMarshalerOption("application/json", newJSONUnsupportedMarshaler()),
```

Comment:
If we just override default `*` marshaler, it will return a **binary** protobuf response with the status message "Unexpected EOF", which might be confusing when users sent JSON payload with `Content-Type: application/json` header, so I've implemented a custom marshaler which returns a better error message.